main.js

```diff
@@ -1 +1,126 @@
+      name,
+      marker,
+    };
+  };
+
+  document.getElementById("startGame").addEventListener("click", function () {
+    const player1name = document.getElementById("player1name").value;
+    const player2name = document.getElementById("player2name").value;
+    const player1marker = document.querySelector(
+      `input[name="marker1"]:checked`
+    ).value;
+    const player2marker = document.querySelector(
+      `input[name="marker2"]:checked`
+    ).value;
+    if (player1name == "") {
+      alert("player 1 name cannot be empty");
+    } else if (player2name == "") {
+      alert("player 2 name cannot be empty");
+    } else if (player1marker == player2marker) {
+      alert("players cannot have same marker");
+    } else {
+      _startGame(player1name, player1marker, player2name, player2marker);
+      document.getElementById("startGame").disabled = true;
+    }
+  });
+
+  _checkWin = function (player1name, player2name) {
+    if (
+      (_gameboard[0] == "X" && _gameboard[1] == "X" && _gameboard[2] == "X") ||
+      (_gameboard[3] == "X" && _gameboard[4] == "X" && _gameboard[5] == "X") ||
+      (_gameboard[6] == "X" && _gameboard[7] == "X" && _gameboard[8] == "X") ||
+      (_gameboard[0] == "X" && _gameboard[3] == "X" && _gameboard[6] == "X") ||
+      (_gameboard[1] == "X" && _gameboard[4] == "X" && _gameboard[7] == "X") ||
+      (_gameboard[2] == "X" && _gameboard[5] == "X" && _gameboard[8] == "X") ||
+      (_gameboard[0] == "X" && _gameboard[4] == "X" && _gameboard[8] == "X") ||
+      (_gameboard[2] == "X" && _gameboard[4] == "X" && _gameboard[6] == "X")
+    ) {
+      alert(`${player1name} you win`);
+      restartGame()
+    } else if (
+      (_gameboard[0] == "O" && _gameboard[1] == "O" && _gameboard[2] == "O") ||
+      (_gameboard[3] == "O" && _gameboard[4] == "O" && _gameboard[5] == "O") ||
+      (_gameboard[6] == "O" && _gameboard[7] == "O" && _gameboard[8] == "O") ||
+      (_gameboard[0] == "O" && _gameboard[3] == "O" && _gameboard[6] == "O") ||
+      (_gameboard[1] == "O" && _gameboard[4] == "O" && _gameboard[7] == "O") ||
+      (_gameboard[2] == "O" && _gameboard[5] == "O" && _gameboard[8] == "O") ||
+      (_gameboard[0] == "O" && _gameboard[4] == "O" && _gameboard[8] == "O") ||
+      (_gameboard[2] == "O" && _gameboard[4] == "O" && _gameboard[6] == "O")
+    ) {
+      alert(`${player2name} you win`);
+      restartGame()
+    } else if (movesPlayed == 9){
+      alert('tie game');
+      restartGame()
+    }
+  };
+
+  const _startGame = (
+    player1name,
+    player1marker,
+    player2name,
+    player2marker
+  ) => {
+    let player1 = _playerGenerator(player1name, player1marker);
+    let player2 = _playerGenerator(player2name, player2marker);
+    const playarea = document.getElementById("playarea");
+    playarea.addEventListener("click", function (e) {
+      if (turn == "player1" && e.target.innerText == player2.marker) {
+        alert("must pick empty square");
+        turn = "player1";
```

Comment:
why do we need to set the turn here?

Reply:
your right that was un-needed as turn is already player 1 and it has not changed in this instance. Editing now